mock queryRenderedFeatures

mcwhittemore opened the pull request:
adds simple user based mocking for queryRenderedFeatures.

mcwhittemore committed:
mock queryRenderedFeatures

mcwhittemore committed:
very weak implementation of queryRenderedFeatures...

mcwhittemore commented:
@davidtheclark can you give this a review?

davidtheclark commented on classes/map.js:
```diff
@@ -92,6 +97,37 @@ Map.prototype.dragPan = {
+        features: [feature]
+      });
+      var merged = turfUnion(fpoly, bpoly);
+      return merged.geometry.type === 'Polygon';
```
I do not understand what is happening with the envelope, union, and filtering out non-Polygons. Could you explain?

mcwhittemore committed:
refactor queryRenderedFeatures to not need envelope

mcwhittemore commented:
@davidtheclark reading the code over, removing envelope should make this a more accurate mock. I have rewritten this and added a handful of comments to try and explain what is going on.

davidtheclark commented:
Cool, looks good. If it's working the way you intend in your Draw tests, then merge away.
